### PR TITLE
AWS ParallelCluster 3 for MPAS

### PR DESCRIPTION
**Pull request by t-brown**
Update the MPAS to ParallelCluster 3.1.2.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/mpas/scripts/setup/create_parallelcluster_config.sh [hunk 1]
```diff
@@ -89,7 +89,7 @@ else
 fi
 
 #Get AWS ParallelCluster Version
-PCLUSTER_VERSION=`pcluster version`
+PCLUSTER_VERSION=`pcluster version | jq -r '.version'`
 
 # Retrieve MPAS Image ID
 MPAS_AMI=`aws ec2 describe-images --owners self \
```
Comment (mhuguesaws) [inline apps/mpas/scripts/setup/create_parallelcluster_config.sh:95]:
```suggestion
export MPAS_AMI=`aws ec2 describe-images --owners self \
```

File: apps/mpas/scripts/cleanup/cleanup_solution_components.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ else
     echo "[INFO] AWS_REGION = ${AWS_REGION}"
 fi
 
-PARALLELCLUSTER_CONFIG="${PARENT_PATH}/../../config/mpas-x86-64.ini"
+PARALLELCLUSTER_CONFIG="${PARENT_PATH}/../../config/mpas-x86-64.yaml"
 
 SSH_KEY_NAME=`crudini --get ${PARALLELCLUSTER_CONFIG} "cluster default" key_name`
```
Comment (mhuguesaws) [inline apps/mpas/scripts/cleanup/cleanup_solution_components.sh:15]:
```suggestion
export SSH_KEY_NAME=`crudini --get ${PARALLELCLUSTER_CONFIG} "cluster default" key_name`
```

**Review by mhuguesaws — COMMENTED** [review]
Thank you for your contribution. I made a couple of comments.

File: apps/mpas/README.md [hunk 3]
```diff
@@ -128,13 +129,13 @@ Create the MPAS Cluster
...
-pcluster create ${CLUSTER_NAME} -c config/mpas-x86-64.ini --region ${AWS_REGION}
+pcluster create-cluster -n ${CLUSTER_NAME} -c config/mpas-x86-64.yaml --region ${AWS_REGION}
 ```
 
 Connect to the cluster
 
 ```bash
-pcluster ssh ${CLUSTER_NAME} -i ~/.ssh/${SSH_KEY_NAME} --region ${AWS_REGION}
+pcluster ssh -n ${CLUSTER_NAME} -i ~/.ssh/${SSH_KEY_NAME} --region ${AWS_REGION}
 ```
```
Comment (mhuguesaws) [inline apps/mpas/README.md:140]: The pcluster delete needs to be updated as well.

File: apps/mpas/README.md [hunk 4]
```diff
@@ -69,7 +69,8 @@ source .env/bin/activate
 Install AWS ParallelCluster
 
 ```bash
-pip3 install aws-parallelcluster==2.11.3
+pip3 install pyrsistent==0.16.1
```
Comment (mhuguesaws) [inline apps/mpas/README.md:72]: What is the package use for ?

**Review by mhuguesaws — COMMENTED** [review]
Left additional comment

File: apps/mpas/README.md (hunk 4 above)
Comment (t-brown) [reply apps/mpas/README.md:72]: It's a dependency of `aws-parallelcluster`, by default it tries to pull in the latest (0.18.1), which fails with a version mismatch.

OK, scratch that. It works and is not required. Let me update the PR.

File: apps/mpas/README.md [hunk 5]
```diff
@@ -204,7 +204,7 @@ To avoid unexpected charges to your account relative to the WRF cluster, make su
 ### Delete the cluster.
 
 ```bash
-pcluster delete ${CLUSTER_NAME} --region ${AWS_REGION}
+pcluster delete -n ${CLUSTER_NAME} --region ${AWS_REGION}
```
Comment (mhuguesaws) [inline apps/mpas/README.md:207]:
```suggestion
pcluster delete-cluster -n ${CLUSTER_NAME} --region ${AWS_REGION}
```

**Review by mhuguesaws — COMMENTED** [review]
Please address comments on deleting the cluster

**Review by mhuguesaws — APPROVED** [review]
LGTM

**Pull request merged**